### PR TITLE
refactor: Updated JS files to TS

### PR DESCRIPTION
**Pull request by adithyaakrishna**
**Description:**

- This PR updates a couple of remaining `.jsx` files to `.tsx`
  - `404.tsx`
  - `500.tsx`
  - `pdf-viewer.tsx`
  - `PDFArrayCustom.ts`
- Also, updates `PDFArrayCustom.ts` from Class Components to Functional Components

### CODE REVIEW

File: apps/web/components/editor/editable-field.tsx [hunk 1]
```diff
@@ -57,7 +58,7 @@ export default function EditableField(props: FieldPropsType) {
       <div
         hidden={props.hidden}
         ref={nodeRef}
-        className="absolute top-0 left-0 m-auto h-16 w-48 cursor-move select-none flex-row-reverse p-2 text-center text-lg font-bold opacity-80"
+        className={`absolute top-0 left-0 m-auto h-16 w-48 cursor-move select-none flex-row-reverse p-2 text-center text-lg font-bold opacity-80 ${props.className}`}
```
Comment (adithyaakrishna) [inline apps/web/components/editor/editable-field.tsx:61]: Not sure if this was the right place to add custom className, please do lmk if I need to make changes to this :)

**Comment by Mythie** [comment]
Hey there, it looks like you haven't accepted our contributor license agreement yet. In order for us to accept your pull request we ask that you please fill out the CLA:

https://documen.so/cla

**Comment by adithyaakrishna** [comment]
Oops! My bad, thanks for reminding me. I have accepted it :)

**Comment by adithyaakrishna** [comment]
Also @Mythie I guess we don't have any CI to check if the PRs actually don't break anything. Would love to know your opinion on adding a `build-test.yml` file to check if it builds successfully

**Comment by adithyaakrishna** [comment]
Hadn't checked the feat/refresh branch, and these files were already moved to TS

**Comment by Mythie** [comment]
Even so we still appreciate the effort you made to do these super valuable updates 🤗 

We're sorry that we hadn't communicated this as well as we could have 💚

**Comment by Mythie** [comment]
> Also @Mythie I guess we don't have any CI to check if the PRs actually don't break anything. Would love to know your opinion on,
> 
> * Adding a `build-test.yml` file to check if it builds successfully
> * A `dependabot.yml` file to update the package versions automatically
> * A `code-ql.yml` file for discovering vulnerabilities
> * And finally a `lint.yml` file to check for linting issues

We are interested in the dependabot and CodeQL aspects of this though 👍

**Comment by adithyaakrishna** [comment]
@Mythie I'll send a PR for those changes then :)

**Pull request closed without merging**